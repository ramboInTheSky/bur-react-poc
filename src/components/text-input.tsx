import React, { FC } from 'react';
import styled from 'styled-components';

const Container = styled.div``;
const Input = styled.input``;

interface Props {
    readonly disabled?: boolean;
    readonly onChange?: ({target}: {target: HTMLInputElement}) => void;
    readonly placeholder?: string;
    readonly type?: string;
    readonly value?: string;
}

export const TextInput: FC<Props> = ({ disabled = false, ...props }) => 
<Container>
    <Input {...props} disabled={disabled} />
</Container>;