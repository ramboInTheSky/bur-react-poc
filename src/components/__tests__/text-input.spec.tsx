import React from 'react';
import { shallow } from 'enzyme';

import { TextInput } from '../text-input';

describe(TextInput, () => {
  it('should render correctly', () => {
    expect(shallow(<TextInput />)).toMatchSnapshot();
  });

  it('should render correctly with a value', () => {
    expect(shallow(<TextInput value="test" />)).toMatchSnapshot();
  });

  it('should render correctly when disabled', () => {
    expect(shallow(<TextInput disabled />)).toMatchSnapshot();
  });
});
