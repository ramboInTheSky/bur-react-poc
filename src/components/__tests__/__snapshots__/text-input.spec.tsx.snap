// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput should render correctly 1`] = `
<styled.div>
  <styled.input
    disabled={false}
  />
</styled.div>
`;

exports[`TextInput should render correctly when disabled 1`] = `
<styled.div>
  <styled.input
    disabled={true}
  />
</styled.div>
`;

exports[`TextInput should render correctly with a value 1`] = `
<styled.div>
  <styled.input
    disabled={false}
    value="test"
  />
</styled.div>
`;
